lib/wibox/layout/grid.lua

```diff
@@ -0,0 +1,629 @@
+-- **Signal:** widget::reset
+-- @param layout The layout you are modifying.
+-- @name reset
+-- @class function
+
+
+--- The preferred orientation for the grid.
+-- @beautiful beautiful.grid_orientation
+
+--- If widget superposition is allowed.
+-- @beautiful beautiful.grid_superpose
+
+--- The spacing between cells.
+-- @beautiful beautiful.grid_[horizontal|vertical]_spacing
+
+--- The space between the grid and the surrounding area.
+-- @beautiful beautiful.grid_[horizontal|vertical]_padding
+
+--- Controls if the columns/rows are expanded to use all the available width/height.
+-- @beautiful beautiful.grid_[horizontal|vertical]_expand
+
+--- Controls if the columns/rows have the same size or if the size depend on the content.
+-- @beautiful beautiful.grid_[horizontal|vertical]_fixed_size
+
+
+--- If false, check before adding a new widget if it will superpose with other widget and prevent from adding it.
+--@DOC_wibox_layout_grid_superpose_EXAMPLE@
+-- @property superpose
+
+--- The spacing between cells.
+-- If direction (horizontal|vertical) is not specified, set both directions and get the preferred orientation
+--@DOC_wibox_layout_grid_spacing_EXAMPLE@
+-- @property [horizontal_|vertical_|]spacing
+
+--- The space between the grid and the surrounding area.
+-- If direction (horizontal|vertical) is not specified, set both directions and get the preferred orientation
+--@DOC_wibox_layout_grid_spacing_EXAMPLE@
+-- @property [horizontal_|vertical_|]padding
+
+--- Controls if the columns/rows are expanded to use all the available width/height.
+-- If direction (horizontal|vertical) is not specified, set both directions and get the preferred orientation
+--@DOC_wibox_layout_grid_expand_EXAMPLE@
+-- @property [horizontal_|vertical_|]expand
+
+--- Controls if the columns/rows have the same size or if the size depend on the content.
+-- If direction (horizontal|vertical) is not specified, set both directions and get the preferred orientation
+--@DOC_wibox_layout_grid_expand_EXAMPLE@
+-- @property [horizontal_|vertical_|]fixed_size
+
+
+
+-- Return the maximum value of a table
+local function max_value(t)
+    local m = 0
+    for _,v in ipairs(t) do
+        if m < v then m = v end
+    end
+    return m
+end
+
+-- Return the sum of the values in the table
+local function sum_values(t)
+    local m = 0
+    for _,v in ipairs(t) do
+        m = m + v
+    end
+    return m
+end
+
+
+--- Find a widget in a widget_table, by matching the coordinates
+-- using the  `row`:`col` coordinates, and the spans `row_span` and `col_span`
+-- @param widgets_table Table of the widgets present in the grid
+-- @param number row Row number for the top left corner of the widget
+-- @param number col Column number for the top left corner of the widget
+-- @tparam number row_span The number of rows the widget spans (default to 1)
+-- @tparam number col_span The number of columns the widget spans (default to 1)
+-- @treturn table of index of widget_table
```

Comment:
If you document a local function that is then turned into a method, you need to add `@function grid:find_widgets_at` or ldoc wont see it.